File: HackerRank/30 days of code/day 24, more linked lists/day_24_more_linked_lists.rb
class Node

    attr_accessor :data,:next

  def initialize data
        @data = data
        @next = nil
    end
end

class Solution
    
    # Immutable HackerRank code above

    def removeDuplicates(head)
        original = head
        while head.next
            head.data == head.next.data ? (head.next = head.next.next) : (head = head.next)
        end
        return original
    end

    # Immutable HackerRank code below

    def insert(head,value)
        p=Node.new(value)
        if head==nil
             head=p
         elsif head.next==nil
             head.next=p
         else  
             current = head
         while current.next != nil
             current = current.next
       end 
       current.next = p
      end
      return head   
   end  

    def display(head)
       current = head
       while current 
          print current.data," "
          current = current.next
       end
   end
end
       
mylist= Solution.new
head=nil
T=gets.to_i
for i in 1..T
    data=gets.to_i
    head=mylist.insert(head,data)
end
head=mylist.removeDuplicates(head)
mylist.display(head)

File: HackerRank/30 days of code/day 24, more linked lists/day_24_more_linked_lists.cs
using System;
using System.Collections.Generic;
class Node
{
    public int data;
    public Node next;
    public Node(int d){
        data=d;
        next=null;
    }
        
}
class Solution {

  // Immutable HackerRank code above

  public static Node removeDuplicates(Node head) {
    Node original = head;
    while (head.next != null) {
        if (head.data == head.next.data)
            head.next = head.next.next;
        else
            head = head.next;
    }
    return original;
  }

  // Immutable HackerRank code below

    public static  Node insert(Node head,int data)
    {
        Node p=new Node(data);
        
        
        if(head==null)
            head=p;
        else if(head.next==null)
            head.next=p;
        else
        {
            Node start=head;
            while(start.next!=null)
                start=start.next;
            start.next=p;
            
        }
        return head;
    }
    public static void display(Node head)
    {
        Node start=head;
        while(start!=null)
        {
            Console.Write(start.data+" ");
            start=start.next;
        }
    }
    static void Main(String[] args) {
    
        Node head=null;
        int T=Int32.Parse(Console.ReadLine());
        while(T-.0){
            int data=Int32.Parse(Console.ReadLine());
            head=insert(head,data);
        }
        head=removeDuplicates(head);
        display(head);
    }
}
    